src/eva/vision/data/datasets/segmentation/monusac.py

```diff
@@ -0,0 +1,239 @@
+            case _:
+                raise ValueError(f"Invalid 'split' value '{self._split}'.")
+
+        return os.path.join(self._root, directory)
+
+    def _export_semantic_label_masks(self) -> None:
+        """Export semantic label masks to disk."""
+        mask_files = [
+            (index, filename.replace(".tif", ".npy"))
+            for index, filename in enumerate(self._image_files)
+        ]
+        to_export = filter(lambda x: not os.path.isfile(x[1]), mask_files)
+        for sample_index, filename in tqdm.tqdm(
+            list(to_export),
+            desc=">> Exporting semantic masks",
+            leave=False,
+        ):
+            semantic_labels = self._get_semantic_mask(sample_index)
+            np.save(filename, semantic_labels)
+
+    def _load_semantic_mask_file(self, index: int) -> npt.NDArray[Any]:
+        """Load a semantic mask file from disk.
+
+        Args:
+            index: Index of the mask file to load.
+
+        Returns:
+            Loaded mask as a numpy array.
+        """
+        mask_filename = self._image_files[index].replace(".tif", ".npy")
+        return np.load(mask_filename)
+
+    def _get_semantic_mask(self, index: int) -> npt.NDArray[Any]:
+        """Builds and loads the semantic label mask from the XML annotations.
+
+        Args:
+            index: Index of the image file.
+
+        Returns:
+            Semantic label mask as a numpy array.
+        """
+        image_path = self._image_files[index]
+        width, height = imagesize.get(image_path)
+        annotation_path = image_path.replace(".tif", ".xml")
+        element_tree = ElementTree.parse(annotation_path)  # nosec
+        root = element_tree.getroot()
+
+        semantic_labels = np.zeros((height, width), "uint8")  # type: ignore[reportCallIssue]
+        for level in range(len(root)):
+            label = [item.attrib["Name"] for item in root[level][0]][0]
+            class_id = self.class_to_idx.get(label, 254) + 1
+            # for the test dataset an additional class 'Ambiguous' was added for
+            # difficult regions with fuzzy boundaries - we return it as 255
+            regions = [item for child in root[level] for item in child if item.tag == "Region"]
+            for region in regions:
+                vertices = np.array(
+                    [(vertex.attrib["X"], vertex.attrib["Y"]) for vertex in region[1]],
+                    dtype=np.dtype(float),
+                )
+                fill_row_coords, fill_col_coords = draw.polygon(
+                    vertices[:, 0],
+                    vertices[:, 1],
+                    (width, height),
+                )
+                semantic_labels[fill_col_coords, fill_row_coords] = class_id
+
+        return semantic_labels
+
+    def _download_dataset(self) -> None:
+        """Downloads the dataset."""
+        self._print_license()
+        for resource in self._resources:
+            if os.path.isdir(self._root):
```

Comment:
looks like this will only download the first dataset (train), and then skip the test because then `self._root` exists

Reply:
ah good point -- fixed!